src/app/core/services/notification.service.ts

```diff
@@ -139,28 +136,33 @@ export class NotificationService {
             fcmNotifications.forEach(this.sendFCMNotification)
           }
           this.storage.set(StorageKeys.LAST_NOTIFICATION_UPDATE, Date.now())
-        }
+        } else Promise.reject()
       })
   }
 
   sendFCMNotification(notification) {
-    FCMPlugin.upstream(
+    return FCMPlugin.upstream(
```

Comment:
Does the upstream function return a Promise ? If not it maybe better to wrap this in a promise based on success or error callbacks.

Reply:
I've fixed it.